agent/acs/client/acs_client_types.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2014-2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package acsclient
+
+import (
+	"reflect"
+
+	"github.com/aws/amazon-ecs-agent/agent/acs/model/ecsacs"
+)
+
+// typeMappings is a map of type-strings (as passed in acs messages as the
+// 'type' field) to a pointer to the corresponding struct type this type should
+// be marshalled/unmarshalled to/from.
+// This list is currently *manually updated* and assumes that the generated
+// struct type-names within the package *exactly match* the type sent by ACS
+// (true so far; careful with inflections)
+var typeMappings map[string]reflect.Type
+
+func init() {
+	// TODO, this list should be autogenerated
```

Comment:
go:generate from agent/acs/model/api/ecsacs-2014-11-13.api.json?  You don't need to re-parse generated code, just understand the model json.  (Not necessary to address now, just an idea for the future.)